assembly/src/assembler/mod.rs

```diff
@@ -366,14 +385,15 @@ impl Assembler {
     fn ensure_procedure_is_in_cache(
         &self,
         proc_id: &ProcedureId,
+        proc_name: &ProcedureName,
         context: &mut AssemblyContext,
     ) -> Result<(), AssemblyError> {
```

Comment:
As mentioned in the previous comments, I would not change the signature of this function but instead would get procedure name from the context inside this procedure only if there is an actual error.